Update release checklist with bumping release version

hickeyma opened the pull request:
**What this PR does / why we need it**:
Add to the release checklist about updating the release version in the master branch when creating a release.  This is so that the correct release version is shown when build from master.

**Special notes for your reviewer**:

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

hickeyma committed:
Update release checklist with bumping release version

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

bacongobbler commented on docs/release_checklist.md:
```diff
@@ -147,6 +147,24 @@ git add .
+git cherry-pick -x <commit-id>
+
+# commit the change
+git push upstream bump-version-<release-version>
```
In this particular case, we probably want to push it to your origin remote instead. The only time we want to push to upstream is for the release branches. :)

hickeyma replied:
Nice catch, thanks. *shaking fist* at copy/paste! :)

bacongobbler reviewed (commented):
One minor thing but LGTM otherwise! Thanks for picking this up

hickeyma committed:
Change command to push to origin remote

Update review comment:
- https://github.com/helm/helm/pull/5712#pullrequestreview-236124918

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma commented:
Thanks for review @bacongobbler
